internal/services/apimanagement/api_management_api_resource.go

```diff
@@ -498,14 +487,184 @@ func resourceApiManagementApiCreateUpdate(d *pluginsdk.ResourceData, meta interf
 		params.Properties.TermsOfServiceURL = pointer.To(v.(string))
 	}
 
-	if err := client.CreateOrUpdateThenPoll(ctx, newId, params, api.CreateOrUpdateOperationOptions{IfMatch: pointer.To("*")}); err != nil {
+	if err := client.CreateOrUpdateThenPoll(ctx, id, params, api.CreateOrUpdateOperationOptions{IfMatch: pointer.To("*")}); err != nil {
 		return fmt.Errorf("creating/updating %s: %+v", id, err)
 	}
 
 	d.SetId(id.ID())
 	return resourceApiManagementApiRead(d, meta)
 }
 
+func resourceApiManagementApiUpdate(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).ApiManagement.ApiClient
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	path := d.Get("path").(string)
+	version := d.Get("version").(string)
+	versionSetId := d.Get("version_set_id").(string)
+	displayName := d.Get("display_name").(string)
+	protocolsRaw := d.Get("protocols").(*pluginsdk.Set).List()
+	protocols := expandApiManagementApiProtocols(protocolsRaw)
+	sourceApiId := d.Get("source_api_id").(string)
+	serviceUrl := d.Get("service_url").(string)
+
+	if version != "" && versionSetId == "" {
```

Comment:
We can reduce code duplication and get the following checks checked at plan time rather than apply time by using doing `CustomizeDiff`